File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) 
    {
        vector<vector<string>>ans;
        unordered_map<string,vector<string>>m;

        for(int i=0;i<strs.size();i++)
        {
            string word=strs[i];
            sort(word.begin(),word.end());
            m[word].push_back(strs[i]);
        }

        for(auto it: m)
        {
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) 
    {
        Map<String,List<String>> mp=new HashMap<>();

        for(int i=0;i<strs.length;i++)
        {
            String str=strs[i];

            char ch[]=str.toCharArray();
            Arrays.sort(ch);
            String key=new String(ch)
;
            if(!mp.containsKey(key))
            mp.put(key,new ArrayList<>());

            mp.get(key).add(str);
        }

        List<List<String>> ans=new ArrayList<>(mp.values());
        // for (List<String> list : ) {
        //     ans.add(list);
        // }

        return ans;
    }
}